content/blog/migrating-a-cloud-application-to-kubernetes/index.md

```diff
@@ -0,0 +1,476 @@
+                                }]
+                            }]
+                        }
+                    }
+                }
+            }
+        },
+        strategy: {
+            type: "Recreate"
+        }
+    }}, {
+    deleteBeforeReplace: true,
+    provider: eksCluster.provider,
+});
+```
+
+In order to have a clean, simple URL through which we can access the database, we create a Kubernetes service for it.
+
+```typescript
+const databasesideListener = new k8s.core.v1.Service("database-side-listener", {
+    metadata: { labels: databaseAppLabels },
+    spec: {
+        type: "LoadBalancer",
+        ports: [{ port: 5432, targetPort: "http" }],
+        selector: databaseAppLabels,
+    }}, {
+    provider: eksCluster.provider,
+    }
+);
+
+const postgresqlAddress = databasesideListener.status.loadBalancer.ingress[0].hostname;
+});
+```
+
+We can now treat our Kubernetes database the same way as we did our RDS one. Like the PERN application, we create a user, schema, and table, grant permissions for our user to edit and select it, and populate the table with two initial voting options.
+
+```typescript
+const postgresqlProvider = new postgresql.Provider("postgresql-provider", {
+        host: postgresqlAddress,
+        port: 5432,
+        username: sqlAdminName,
+        password: sqlAdminPassword,
+        superuser: false,
+        sslmode: "disable",
+});
+
+const postgresDatabase = new postgresql.Database("postgresql-database", {
+    name: "votes"}, {
+    provider: postgresqlProvider,
+});
+
+const postgresUser = new postgresql.Role("postgres-standard-role", {
+    name: sqlUserName,
+    password: sqlUserPassword,
+    superuser: false,
+    login: true,
+    connectionLimit: -1}, {
+    provider: postgresqlProvider,
+});
+
+const creationScript = `
+    CREATE SCHEMA voting_app;
+    CREATE TABLE voting_app.choice(
+        choice_id SERIAL PRIMARY KEY,
+        text VARCHAR(255) NOT NULL,
+        vote_count INTEGER NOT NULL
+    );
+    GRANT USAGE ON SCHEMA voting_app TO ${sqlUserName};
+    GRANT SELECT, UPDATE ON ALL TABLES IN SCHEMA voting_app TO ${sqlUserName};
+    INSERT INTO voting_app.choice (text, vote_count) VALUES('Tabs', 0);
+    INSERT INTO voting_app.choice (text, vote_count) VALUES('Spaces', 0);
+`;
+
+const deletionScript = "DROP SCHEMA IF EXISTS voting_app CASCADE";
+
+const postgresqlVotesSchema = new Schema("postgresql-votes-schema", {
+    creatorName: sqlAdminName,
+    creatorPassword: sqlAdminPassword,
+    serverAddress: postgresqlAddress,
+    databaseName: postgresDatabase.name,
+    creationScript: creationScript,
+    deletionScript: deletionScript,
+    postgresUserName: postgresUser.name,
+});
+```
+
+And now, all that remains is the deploy the client and server containers to Kubernetes. Since the PERN application already deplyed them as docker containers, it doesn't take too much effort to reconfigure it for Kubernetes.
```

Comment:
```suggestion
And now, all that remains is to deploy the client and server containers to Kubernetes. Since the PERN application already deployed them as docker containers, it doesn't take too much effort to reconfigure it for Kubernetes.
```